### DISCUSSION TITLE
Drawing a signature and 2.1.1 Keyboard checkpoint

### DISCUSSION THREAD

**Discussion by AccessibilityEnthusiastic** [w3c/wcag#2742] [Q&A] [answered]
Hello team,

I am wondering if drawing a signature is a failure of WCAG 2.1.1 Keyboard

Is it required to be accessible by keyboard?

As per WCAG example 1:

Example 1: A drawing Program.

A drawing program allows users to create, size, position and rotate objects from the keyboard
https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html#examples

so can we consider it as a failure of 2.1.1?
If yes then is there any suggestion to make the signature editing accessible by keyboard? Please suggest!!
![Signature](https://user-images.githubusercontent.com/115781693/196921129-78d59f85-5aac-4b1d-8c3c-adf9ba10368d.png)

**Comment by bruce-usab** [comment]
@AccessibilityEnthusiastic &ndash; I am trying a new (to me) feature with moving your question from issues to discussions, as I do not think you are requesting a change to WCAG or Understanding et al.

Since the context is a signature, i.e. handwriting, I think the note is applicable:

> This exception relates to the underlying function, not the input technique. For example, if using handwriting to enter text, the input technique (handwriting) requires path-dependent input but the underlying function (text input) does not.

Unless the web page is performing live signature matching (doubtful) IMHO this sort of thing is an unambiguous failure against 2.1.1.

It used to be more of an issue, but nowadays digital signatures (which can be accessible) have largely replaced asking end-users to pretend a mouse is like a pencil.  For forms without that option, for years now, end-users can upload a picture (presumably of their signature) or just type in their name.

**Comment by patrickhlauke** [comment, marked as answer]
I'll also add that most online solutions like DocuSign and similar offer a basic text entry equivalent (with a choice of pseudo-handwriting fonts) to sign documents (which always makes me wonder what the legal enforceability here would be to prove that it was indeed you who signed it, but that's a separate issue).

![screenshot of a DocuSign signing dialogue, where users can type in their name/initials and choose from a selection of handwriting-like fonts](https://user-images.githubusercontent.com/895831/196946040-a98c8ef2-0ea5-41ff-95c0-413e5d59efa2.png)

**Comment by patrickhlauke** [comment]
To expand on my (old) answer above: the act of drawing the signature itself does not need to be keyboard-accessible (as it would be a ridiculous requirement for a keyboard user to have to "draw" their signature). what does need to be keyboard-accessible is the reason why a page/site asks for a signature, i.e. the fact that it's asking for explicit confirmation from the user in order to proceed.
